### ISSUE TITLE
Ticket Item grid allows invalid values into the model

### ISSUE THREAD

**Issue by githubbob42** [githubbob42/mingle2github2#4304]
**[Mingle Card: 4641](https://liquidframeworks.mingle.thoughtworks.com/projects/alpine_mobile/cards/4641)**
Steps to Reproduce
------------------

```
| | |
|-|-|
|**Version #**| |
|**Hardware**| |
|**OS**| |
|**Browser**| |
|**Username**| |
|**Password**|Use LastPass|
|ORG ID| |
|User ID| |
|RayGun Error ID| |
  
 

```

1.  Setup min Qty to 5 for a pbItem in console
2.  Sync, and on a ticket, add the pbItem
3.  Change the input Qty to 3
4.  Turn Edit mode off

Expected Result
---------------

_value shouldn’t persist_

Actual Result
-------------

_value persists_

_Analysis_
----------

Maurya found this issue BEFORE 

After changing the errorDisplayField property on the rule from Qty\_\_c (readonly/formula) to Input\_Qty\_\_c the issue goes away.  Now, this is not a solution but a smell to the issue.  The errorDisplayField property is a display/layout concern that determines WHERE the error is displayed and shouldn’t affect the running of rules.  

It appears that we are using the errorDisplayField property to determine which rules get run.  Not good.  All rules for an object should be run regardless of where they are displayed.

Line 77 of layout/form.js is only adding rules to fields if the field name matches the errorDisplayField which means only rules with error display fields are run.  Lines 83-88 of viewmodels/form-validation.js were added later, apparently in an effort to correct the issue created in form.js by filtering the rules by errorDisplayField.

Another issue with this approach is that is filters out invalid rules.  E.g. rules for fields the no longer exist (or perhaps a field you don’t have access to?).  Based on the original requirements/use-cases, any rule that fails to evaluate is treated as a failed rule.  This covers rules with typos, bad syntax or invalid fields, etc.

Related Cards
-------------

Card #4661

#4675

Test Plan
---------

We found that this easily breaks with the following rule:

```
      'Min\_Max\_Qty\_Enforcement' : {
        'fullName' : 'Min\_Max\_Qty\_Enforcement',
        'errorConditionFormula' : '( ( (FX5\_\_Input\_Qty\_\_c  < NULLVALUE(FX5\_\_PBI\_Minimum\_Quantity\_\_c, 0)) && NOT(ISNULL(FX5\_\_PBI\_Minimum\_Quantity\_\_c))) || ( (FX5\_\_Input\_Qty\_\_c  > NULLVALUE( FX5\_\_PBI\_Maximum\_Quantity\_\_c, 0)) && NOT(ISNULL(FX5\_\_PBI\_Maximum\_Quantity\_\_c))) )',
        'errorDisplayField' : 'Qty\_\_c',
        'errorMessage' : 'The quantity must be greater than or equal to the minimum quantity and less than or equal to the maximum quantity.'
      },


```

Note that the errorDisplayField is set to Qty\_\_c, when we set it to FX5\_\_Input\_Qty\_\_c the validation worked.

**Comment by githubbob42** [comment]
Matthew Todd : 12/12/2016 13:22 CST
Test Result: Passed PR Build: 4.0.1.pr2146 on 8.0.0 Admin User: qaadmin@test4.com Field User: qafield@test4.com Hardware: Surface Pro 3 OS: Windows 10 Pro Browser: Chrome Incognito 55.0.2883.75 m (64-bit) URL: https://hunterpeak.herokuapp.com Test Case: Validation Rules on Ticket Items Grid (4611, 4639) Test Notes: Invalid values are no longer being stored in the model. When coming out of 'Edit' mode, the values are still retained, but performing a refresh by leaving and coming back to the page shows the true values at their previously correct value. The retained values after coming out of Edit are being addressed in #4675.

**Comment by githubbob42** [comment]
**Pull Request #2146**
